internal/controller/kyma_controller.go

```diff
@@ -371,6 +390,7 @@ func (r *KymaReconciler) handleProcessingState(ctx context.Context, kyma *v1beta
 	}
 
 	if err := errGroup.Wait(); err != nil {
+		r.Metrics.RecordRequeueReason(requeueReason)
```

Comment:
this `Metrics.RecordRequeueReason` should not be put here, there has race condition in each errGroup.Go routine, you'll lost some error, for example, the `ManifestReconciliationError` and `ModuleCatalogSyncError` may happens during same time period, however, the err returned here is the first one get returned by those two threads (goroutine), and you will lost another error.

Instead, put RecordRequeueReason(requeueReason) in each errGroup goroutine.